Fixes #26105 - Configurable error severity

xprazak2 opened the pull request:
```ruby
render :json => { :error => { :errors => { :base => messages }, :severity => severity } }, :status => 422
```
when severity is `info`,`danger`, or `warning`

![form-info](https://user-images.githubusercontent.com/2664090/53175159-b1516900-35eb-11e9-93da-246a696ec3bd.png)

![form-danger](https://user-images.githubusercontent.com/2664090/53175162-b31b2c80-35eb-11e9-9f12-646b38f74f3c.png)

![form-warn](https://user-images.githubusercontent.com/2664090/53175164-b4e4f000-35eb-11e9-8b16-de2e8f61c3ad.png)

Commit:
Fixes #26105 - Configurable error severity

tbrisker commented on webpack/assets/javascripts/react_app/components/common/forms/Form.js:
```diff
@@ -20,9 +20,9 @@ const Form = ({
   <form className={className} onSubmit={onSubmit}>
     {error && (
-      <Alert className="base in fade" type="danger">
+      <Alert className="base in fade" type={error.severity}>
```
Would it be makes sense to handle the default here instead of in the form component? something like
```suggestion
      <Alert className="base in fade" type={error.severity || "danger"}>
```

tbrisker replied:
Looks like this change was somehow lost in the rebases?

xprazak2 replied:
It is back.

tbrisker reviewed (changes requested):
see comment inline

ohadlevy reviewed (dismissed):
looking good, but it seems some snapshot failed, can you please have a look? thanks.

tbrisker reviewed (dismissed):
Ack pending snapshot fix

xprazak2 commented:
I fixed the snapshots and added minor visual tweak: When there is just one error, the message is aligned with error title:

![](https://user-images.githubusercontent.com/2664090/54489014-43c1f100-48a8-11e9-9a81-3c8374893e0e.png)

![](https://user-images.githubusercontent.com/2664090/54489017-4886a500-48a8-11e9-9035-e009bdefed1f.png)

xprazak2 commented:
@tbrisker, @ohadlevy, any additional comments?

tbrisker commented:
@xprazak2 could you rebase on latest develop to get tests passing please?

xprazak2 commented:
Rebased.

mmoll commented:
all :green_heart: now

ohadlevy commented on webpack/assets/javascripts/react_app/components/common/forms/Form.js:
```diff
@@ -52,7 +57,7 @@ Form.defaultProps = {
   disabled: false,
   submitting: false,
-  errorTitle: __('Unable to save'),
+  errorTitle: __('Unable to save. '),
```
nitpick: this has extra whitespace, also not sure if the dot is really required (as its a title?)

xprazak2 replied:
The dot and whitespace is intentional to have nice formatting if there is only [1 error](https://github.com/theforeman/foreman/pull/6499#issuecomment-473653134)

ares replied:
that may break during translations, it would be better to do something like `__('Unable to save') + ". "` I think

xprazak2 replied:
Updated.

Commit:
Refs #26105 - Update default error message

Commit:
Refs #26105 - Return back the missing change

tbrisker reviewed (dismissed):
Thanks @xprazak2 !

tbrisker commented:
Travis failure related

xprazak2 commented:
Travis finally green again.

tbrisker reviewed (approved):
thanks @xprazak2 !
